File: app/src/main/java/com/aiwujie/shengmo/adapter/SearchAdapter.java
package com.aiwujie.shengmo.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.aiwujie.shengmo.R;
import com.aiwujie.shengmo.bean.AddressBean;

import java.util.List;

/**
 * Created by 290243232 on 2016/12/22.
 */
public class SearchAdapter extends BaseAdapter {
    private List<AddressBean> data;
    private LayoutInflater li;
    public SearchAdapter(Context context){
        li = LayoutInflater.from(context);
    }

    /**
     * 设置数据集
     * @param data
     */
    public void setData(List<AddressBean> data){
        this.data = data;
    }

    @Override
    public int getCount() {
        return  (data==null)? 0: data.size();
    }
    @Override
    public Object getItem(int position) {
        return data.get(position);
    }
    @Override
    public long getItemId(int position) {
        return position;
    }
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder vh = null;
        if(convertView == null){
            vh = new ViewHolder();
            convertView = li.inflate(R.layout.item_map_address, null);
            vh.title = (TextView) convertView.findViewById(R.id.map_item_title);
            vh.text = (TextView) convertView.findViewById(R.id.map_item_text);
            convertView.setTag(vh);
        }else{
            vh = (ViewHolder) convertView.getTag();
        }
        vh.title.setText(data.get(position).getTitle());
        vh.text.setText(data.get(position).getProvince()+data.get(position).getCity()+data.get(position).getText());
        return convertView;
    }

    class ViewHolder{
        public TextView title;
        public TextView text;
    }
}

File: app/src/main/java/com/aiwujie/shengmo/kt/adapter/SearchAdapter.kt
package com.aiwujie.shengmo.kt.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.aiwujie.shengmo.R
import com.aiwujie.shengmo.bean.AddressBean
import com.aiwujie.shengmo.kt.util.showToast

class SearchAdapter(var context: Context, var list: List<AddressBean> =ArrayList())  : RecyclerView.Adapter<SearchAdapter.SearchAdapterViewHolder>() {



    inner class SearchAdapterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvTitle: TextView = itemView.findViewById(R.id.map_item_title)
        val tvText: TextView = itemView.findViewById(R.id.map_item_text)

        fun display(position: Int) {
            list[position].run {
                tvTitle.text = title
                tvText.text = province + city + text

              itemView.setOnClickListener {
                  mOnAddressClick!!.onClick(it,position)
              }


            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): SearchAdapterViewHolder {
        return SearchAdapterViewHolder(LayoutInflater.from(context).inflate(R.layout.item_map_address, parent, false))
    }

    override fun getItemCount(): Int {
        return list.size

    }

    override fun onBindViewHolder(holder: SearchAdapterViewHolder?, position: Int) {
        holder?.display(position)

    }


    var mOnAddressClick: OnAddressClick? = null

    interface OnAddressClick {
        fun onClick(view:View,position: Int)
    }

    fun setOnAddressClick(onAddressClick: OnAddressClick?) {
        mOnAddressClick = onAddressClick
    }


}